### PR TITLE
Retry Service Dependencies on Startup -- ExportDistro

### PR DESCRIPTION
**Pull request by brandonforster**
#442

### CODE REVIEW

**Comment by jduranf** [comment]
@brandonforster please run `make test` before commit

File: cmd/export-distro/main.go [hunk 1]
```diff
@@ -11,67 +11,70 @@ package main
...
 
+var bootTimeout = 30000 //Once we start the V2 configuration rework, this will be config driven
+
 func main() {
-	var (
-		useConsul  bool
-		useProfile string
-	)
+	start := time.Now()
+	var useConsul bool
```
Comment (drasko) [inline cmd/export-distro/main.go]: Group vars and put them before `start`.

File: cmd/export-distro/main.go [hunk 2]
```diff
@@ -11,67 +11,70 @@ package main
...
 	flag.Parse()
 
-	configuration := &distro.ConfigurationStruct{}
-	err := config.LoadFromFile(useProfile, configuration)
-	if err != nil {
-		logBeforeInit(fmt.Errorf("%s: version: %s: err: %s", internal.ExportDistroServiceKey, edgex.Version, err.Error()))
+	params := startup.BootParams{UseConsul: useConsul, UseProfile: useProfile, BootTimeout: bootTimeout}
+	startup.Bootstrap(params, distro.Retry, logBeforeInit)
+
+	ok := distro.Init()
```
Comment (drasko) [inline cmd/export-distro/main.go]: ove definition into `if`, like `if ok :=...; !ok {...`

File: internal/export/distro/init.go [hunk 3]
```diff
@@ -87,14 +128,29 @@ func Init(conf ConfigurationStruct, useConsul bool) error {
...
+	//We currently have to load configuration from filesystem first in order to obtain ConsulHost/Port
+	conf := &ConfigurationStruct{}
+	err := config.LoadFromFile(useProfile, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	if useConsul {
+		err := connectToConsul(conf)
+		if err != nil {
```
Comment (drasko) [inline internal/export/distro/init.go]: move `err` definition to `if err := ...; err != nil {`

File: internal/export/distro/init.go [hunk 4]
```diff
@@ -72,11 +116,8 @@ func ConnectToConsul(conf ConfigurationStruct) error {
 	return nil
 }
 
-func Init(conf ConfigurationStruct, useConsul bool) error {
-	configuration = conf
-	LoggingClient = logger.NewClient(internal.ExportDistroServiceKey, conf.EnableRemoteLogging, getLoggingTarget(conf))
-
-	coreDataEventURL := "http://" + conf.DataHost + ":" + strconv.Itoa(conf.DataPort) + EventUriPath
+func initializeClient(useConsul bool) {
+	coreDataEventURL := "http://" + Configuration.DataHost + ":" + strconv.Itoa(Configuration.DataPort) + EventUriPath
```
Comment (drasko) [inline internal/export/distro/init.go]: Use `fmt.Sprintf()` to be consistent (if used below)

File: internal/export/distro/init.go [hunk 5]
```diff
@@ -87,14 +128,29 @@ func Init(conf ConfigurationStruct, useConsul bool) error {
 	}
 
 	ec = coredata.NewEventClient(params, startup.Endpoint{})
+}
 
-	return nil
+func initializeConfiguration(useConsul bool, useProfile string) (*ConfigurationStruct, error) {
+	//We currently have to load configuration from filesystem first in order to obtain ConsulHost/Port
+	conf := &ConfigurationStruct{}
+	err := config.LoadFromFile(useProfile, conf)
```
Comment (drasko) [inline internal/export/distro/init.go]: move `err` to `if`

File: cmd/export-distro/main.go (hunk 2 above)
Comment (brandonforster) [reply cmd/export-distro/main.go]: `if ok := distro.Init()` doesn't work because the assignment function doesn't return a value.

**Comment by brandonforster** [comment]
Should be merged after PR #580

**Comment by tsconn23** [comment]
@brandonforster PR #580 has been merged. Let's get this one wrapped up.

**Comment by tsconn23** [comment]
recheck

**Pull request merged**